### PR TITLE
symmetrica: 2.0 -> 3.0.1

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

This switches to the sagemath fork, since the original upstream is no
longer maintained and sagemath has effectively maintained symmetrica
through patches for a while now. The update fixes one bug in particular
that has caused failures in the sagemath test suite:

https://github.com/NixOS/nixpkgs/issues/81449
https://trac.sagemath.org/ticket/15312
https://trac.sagemath.org/ticket/29061

Regarding the licensing change:
    
https://gitlab.com/sagemath/symmetrica/-/commit/7cf91b380078a4def28dd6bab2b4607e2346c650

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by timokau** [comment]
Sorry, I lazily used `git rebase -i HEAD~30` to squash a few commits and it looks like github attributed all 30 commits to me. Should be fixed now.

File: pkgs/applications/science/math/symmetrica/default.nix [hunk 1]
```diff
@@ -1,62 +1,36 @@
...
+  meta = with lib; {
     description = ''A collection of routines for representation theory and combinatorics'';
-    license = stdenv.lib.licenses.publicDomain;
-    maintainers = [stdenv.lib.maintainers.raskin];
-    platforms = stdenv.lib.platforms.unix;
-    homepage = http://www.algorithm.uni-bayreuth.de/en/research/SYMMETRICA/index.html;
+    license = licenses.isc;
+    maintainers = with maintainers; [raskin timokau];
+    platforms = platforms.unix;
+    homepage = https://gitlab.com/sagemath/symmetrica;
```
Comment (jonringer) [inline pkgs/applications/science/math/symmetrica/default.nix]:
```suggestion
    homepage = "https://gitlab.com/sagemath/symmetrica";
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

File: pkgs/applications/science/math/symmetrica/default.nix (hunk 1 above)
Comment (timokau) [reply pkgs/applications/science/math/symmetrica/default.nix]: Right, fixed. Thanks!

**Comment by timokau** [comment]
Fast-tracking this a bit since it unbreaks the sage package. We can always make changes or even revert in another PR. I hope you don't mind @7c6f434c.

**Pull request merged**